### PR TITLE
NMS-15130: Added event command

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-15130

### CODE REVIEW

File: features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import org.opennms.netmgt.model.OnmsSeverity;
+
+@Command(scope = "opennms", name = "event", description = "Show event details for a given id")
+@Service
+public class EventCommand implements Action {
+
+    @Reference
+    public EventDao eventDao;
+
+    @Argument(name = "id", description = "Event Id to match (exact).", required = false, multiValued = false)
```
Comment (synqotik) [inline features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java:56]: Should `required` be true? Or else, we should check in `execute()` if it's `null`.

File: features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        fillTable(eventPropertyTable, "eventSnmpHost", onmsEvent.getEventSnmpHost());
+        fillTable(eventPropertyTable, "serviceType", onmsEvent.getServiceType());
+        fillTable(eventPropertyTable, "eventSnmp", onmsEvent.getEventSnmp());
+        fillTable(eventPropertyTable, "eventCreateTime", onmsEvent.getEventCreateTime());
+        fillTable(eventPropertyTable, "eventDescr", onmsEvent.getEventDescr());
+        fillTable(eventPropertyTable, "eventLog", onmsEvent.getEventLog());
+        fillTable(eventPropertyTable, "eventLogGroup", onmsEvent.getEventLogGroup());
+        fillTable(eventPropertyTable, "eventLogMsg", onmsEvent.getEventLogMsg());
+        fillTable(eventPropertyTable, "eventSeverity", onmsEvent.getEventSeverity());
+        fillTable(eventPropertyTable, "ifINdex", onmsEvent.getIfIndex());
```
Comment (synqotik) [inline features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java:100]: `ifINdex` -> `ifIndex`

File: features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        final var onmsEvents = eventDao.findMatching(criteriaBuilder.toCriteria());
+
+        if (onmsEvents.size() == 0) {
+            return null;
+        }
+
+        final OnmsEvent onmsEvent = onmsEvents.get(0);
+
+        final ShellTable eventPropertyTable = new ShellTable();
+        eventPropertyTable.size(terminal.getWidth() - 1);
```
Comment (synqotik) [inline features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java:81]: If `terminal` is `optional`, could it be null here?

File: features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java (hunk 2 above)
Comment (mmahacek) [reply features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java:100]:
```suggestion
        fillTable(eventPropertyTable, "ifIndex", onmsEvent.getIfIndex());
```

File: features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java (hunk 1 above)
Comment (mmahacek) [reply features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java:56]: Unless we say that if `id==null` then we display the most recent event in the database.

Comment (synqotik) [reply features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java:56]: Yes, but then should also change the `description`, as well as fix the `.eq` clause.

Comment (christianpape) [reply features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java]: done

File: features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java (hunk 2 above)
Comment (christianpape) [reply features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java]: done

File: features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java (hunk 3 above)
Comment (christianpape) [reply features/events/shell-commands/src/main/java/org/opennms/netmgt/events/commands/EventCommand.java:86]: done

**Review by synqotik — APPROVED** [review]
LGTM!

**Review by RangerRick — APPROVED** [review]

**Pull request merged**